opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -1902,54 +1905,61 @@ private void indexParallel(String dir, IndexDownArgs args) {
+                                // Allow one retry if interrupted
+                                if (++tries <= 1) {
+                                    continue;
+                                }
+                                LOGGER.log(Level.WARNING, "No retry: ''{0}''", x.file);
+                                x.exception = e;
+                                x.ret = false;
+                            } catch (RuntimeException | IOException e) {
+                                String errmsg = String.format("ERROR addFile(): '%s'", x.file);
+                                LOGGER.log(Level.WARNING, errmsg, e);
+                                x.exception = e;
+                                x.ret = false;
+                            } finally {
+                                if (pctags != null) {
+                                    pctags.reset();
+                                    ctagsPool.release(pctags);
+                                }
                             }
-                        } catch (AlreadyClosedException e) {
-                            alreadyClosedCounter.incrementAndGet();
-                            String errmsg = String.format("ERROR addFile(): '%s'", x.file);
-                            LOGGER.log(Level.SEVERE, errmsg, e);
-                            x.exception = e;
-                            ret = false;
-                        } catch (InterruptedException e) {
-                            // Allow one retry if interrupted
-                            if (++tries <= 1) {
-                                continue;
-                            }
-                            LOGGER.log(Level.WARNING, "No retry: ''{0}''", x.file);
-                            x.exception = e;
-                            ret = false;
-                        } catch (RuntimeException | IOException e) {
-                            String errmsg = String.format("ERROR addFile(): '%s'", x.file);
-                            LOGGER.log(Level.WARNING, errmsg, e);
-                            x.exception = e;
-                            ret = false;
-                        } finally {
-                            if (pctags != null) {
-                                pctags.reset();
-                                ctagsPool.release(pctags);
-                            }
-                        }
 
-                        progress.increment();
-                        return ret;
-                    }
-                }))).get();
+                            progress.increment();
+                            return x;
+                        }
+                    }).
+                    collect(Collectors.toSet());
+            List<Future<IndexFileWork>> futures = parallelizer.getIndexWorkExecutor().invokeAll(callables);
+            for (var future : futures) {
+                IndexFileWork work = future.get();
+                bySuccess.merge(work.ret, new ArrayList<>(List.of(work)), (x, y) -> {
```

Comment:
Can we replace merge with computeIfAbsent
 bySuccess.computeIfAbsent(work.ret, key -> new ArrayList<>())
                 .add(work)

Reply:
bySuccess hashmap is only used to derive the number of failed workers, can we replace this with a simple counter to count the number of failed workers

Reply:
This is more elegant, thanks.